File: C++/armstrong-number.cpp
// Time:  O(klogk)
// Space: O(k)

class Solution {
public:
    bool isArmstrong(int N) {
        const auto& n_str = to_string(N);
        return accumulate(n_str.cbegin(), n_str.cend(), 0,
                          [&](const auto& x, const auto& y) {
                              return x + pow(y - '0', n_str.length());
                          }) == N;
    }
};

File: Python/armstrong-number.py
# Time:  O(klogk)
# Space: O(k)

class Solution(object):
    def isArmstrong(self, N):
        """
        :type N: int
        :rtype: bool
        """
        n_str = str(N)
        return sum(int(i)**len(n_str) for i in n_str) == N
